internal/hooks/v0hooks/manager.go

```diff
@@ -125,6 +129,23 @@ func (o *Manager) invokeHook(
 		}
 		return o.dispatch(
 			r.Context(), &o.config.Hook.CustomAccessToken, conn, input, output)
+
+	case *BeforeUserCreatedInput:
+		if _, ok := output.(*BeforeUserCreatedOutput); !ok {
+			return apierrors.NewInternalServerError(
+				"output should be *hooks.BeforeUserCreatedOutput")
+		}
+		return o.dispatch(
+			r.Context(), &o.config.Hook.BeforeUserCreated, conn, input, output)
+
+	case *AfterUserCreatedInput:
+		_, ok := output.(*AfterUserCreatedOutput)
+		if !ok {
+			return apierrors.NewInternalServerError(
+				"output should be *hooks.AfterUserCreatedOutput")
+		}
```

Comment:
These errors should be `panic` honestly, it's our programmer error. I don't know why they were created like this in the past... (no preference, just a note)

Reply:
My preference here is still to remove the need for these checks all together by having the Input and Output as part of the method signatures instead of everything using InvokeHook. Does seeing this code make you more open to that idea in a future pull request?

Reply:
Yea ofc it does, was just commenting on it for future consideration